buildbot-host/aptly/publish-debian-snapshot.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/sh
+#
+# Publish a package to an aptly repository
+#
+set -e
+
+usage() {
+    echo "Usage: aptly-publish.sh -f <file> -p <passphrase-file> [-s <hook script>] [-h]"
+    echo
+    echo "Options:"
+    echo "    -f    the debian (.deb) package to publish"
+    echo "    -p    GPG passphrase file"
+    echo "    -s    hook script to run"
+    echo "    -h    show this help"
+    echo
+    echo "The repository name (e.g. debian-12) is passed to the hook script as the first"
+    echo "parameter."
+    echo
+    exit 2
+}
+
+HOOK_SCRIPT=""
+
+while getopts 'f:p:s:h' arg
+do
+  case $arg in
+    p) PACKAGE=$OPTARG ;;
+    f) PASSPHRASE_FILE=$OPTARG ;;
```

Comment:
from the usage point of view it looks like the PASSPHRASE_FILE should be with `-p` option, not `-f`?

Reply:
> from the usage point of view it looks like the PASSPHRASE_FILE should be with `-p` option, not `-f`?

I noticed that help text for the options was wrong due to refactoring so that's been fixed.

I think the parameter names will suck either way I think, because *-p* would be natural for both a passphrase file and package. I could add long options (e.g. *--passphrase-file* and *--package*) which would make it more obvious what the options refer to. What do you think?


That said, these scripts are used mainly by Buildbot so the human user interface might not matter as much as elsewhere.

Reply:
> I could add long options (e.g. _--passphrase-file_ and _--package_) which would make it more obvious what the options refer to. What do you think?

I take that back. Bash getopts does not support long options directly so probably not worth the extra effort.

Reply:
yeah, I guess just have help text up to date is enough